provider/bluecat/bluecat.go

```diff
@@ -1008,3 +1045,24 @@ func expandZone(zone string) string {
 	}
 	return ze
 }
+
+//Function to parse the owner from the TXT record
+//It returns in the format externaldns/owner:"OWNER_NAME"
+func extractOwnerfromTXTRecord(propString string) (string, error) {
+	if len(propString) == 0 {
+		return "", errors.Errorf("External-DNS Owner not found")
+	}
+	propsTemp := strings.TrimSuffix(propString, "|")
+	splits := strings.Split(propsTemp, "|")
+	if strings.Contains(strings.Join(splits, ""), "external-dns") {
+		items := strings.Split(strings.Join(splits, ""), ",")
+		var owner = ""
+		for _, value := range items {
+			if strings.Contains(value, "owner") {
+				owner = strings.Split(value, "=")[1]
+			}
+		}
+		return owner, nil
+	}
```

Comment:
Let try and use a [regexp](https://golang.org/pkg/regexp/) to parse the string.

Reply:
Done.